client/src/utils/parser/robotCodeToSsotParsing/robotCodeToSsotParsing.js

```diff
@@ -166,91 +165,56 @@ const currentLineWithoutOutputVariableName = (
  * @returns Array of Objects with the following schema:
  *      instructionBlocks = [rpaApplication:String, rpaTask:String, name:String, paramArray:Array]
  */
-const getInstructionBlocksFromTaskSection = (
-  robotCodeTaskSection,
-  declaredApplications
-) => {
-  let currentApplication;
+const getInstructionBlocksFromTaskSection = (robotCodeTaskSection, taskAndApplicationCombinations) => {
   let errorWasThrown;
   const instructionBlocks = [];
-  const regexForElementNameLine = /#/;
   const regexForOutputVariable = /\${(.)+} =/;
   const splitPlaceholder = '§&§';
 
   robotCodeTaskSection.slice(1).forEach((line) => {
     if (errorWasThrown) return;
-    let currentLine = line.trim();
-    const currentLineDefinesNewApplication = declaredApplications.includes(
-      currentLine
-    );
-    const currentLineContainsElementName = regexForElementNameLine.test(
-      currentLine
-    );
-    const currentLineIncludesSplitPlaceholder = currentLine.includes(
-      splitPlaceholder
-    );
-    const currentLineHasNoSpecifiedApplication =
-      typeof currentApplication === 'undefined';
-    const currentLineDefinesOutputValue = regexForOutputVariable.test(
-      currentLine
-    );
+    let currentLine = line;
```

Comment:
According to an ESlint rule it is not allowed to modify passed parameters. However, because I change the value of the parameter in the function (among other things to cut off the front part of the current line), I need to adjust this string. Therefore I have created a copy of the passed string in the selected line.

![image](https://user-images.githubusercontent.com/36270527/116881609-c1315280-ac23-11eb-968d-0712dbb792c6.png)